### DISCUSSION TITLE
How to Insert Multiple nested Objects

### DISCUSSION THREAD

**Discussion by lizzie-tersano** [prisma/prisma#18332] [Q&A] [answered]
Hello, I am having difficulty understanding how to insert an array of objects properly using prisma.

prisma: 4.11.0
database: mongodb
framework: nestjs

Here is the partial schema:

```prisma
type DeviceParts {
    partNumber          String
    quantity                Int
    partName             String
}

model DeviceModel {
    id                           String      @id @default(auto()) @map("_id") @db.ObjectId
    modelNumber      String      @unique
    parts                     DeviceParts[]
}
```

I have attempted to write this: 

```ts
const newDevice = await this.prismaService.deviceModel.create({
            data: {
                modelNumber: device.modelNumber,
                parts: {
                    createMany: {
                        data: [
                            {partNumber: '', quantity: 1, partName: ''}
                        ],
                    },
                }
            }
        })
```

However, the ide is not allowing me to add a createMany within this "parts" field. Any suggestions on what to do? Thanks.

**Comment by ludralph** [comment, marked as answer]
Hi @lizzie-tersano 👋 

You can try inserting using this code
```js
const newDevice = await prisma.deviceModel.create({
  data: {
    modelNumber: 'A123',
    parts: {
      set: [
        {
          partNumber: '123',
          quantity: 1,
          partName: 'Part A'
        },
        {
          partNumber: '456',
          quantity: 2,
          partName: 'Part B'
        }
      ]
    }
  }
})
```
Composite types can be created within a create or createMany method using the set operation. Take a look at this [section](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types#creating-records-with-composite-types-using-create-and-createmany) of the documentation

**Reply by lizzie-tersano** [reply]
Thanks! Though I'm wondering if you know any good way to do this in a programmatic manner rather than inserting 100+ array items. I attempted to do `set: device.parts` (where parts is an array of DeviceParts) but it won't accept it.

**Reply by ludralph** [reply]
Doing this works for me
```js
 const device = {
    parts : [
      {
        partNumber: '123',
        quantity: 1,
        partName: 'Part A'
      },
      {
        partNumber: '456',
        quantity: 2,
        partName: 'Part B'
      }
    ]
  }

const newDevice = await prisma.deviceModel.create({
  data: {
    modelNumber: 'A1234',
    parts: {
      set: device.parts
    }
  }
})
```
Can you share your code snippet with me?

**Reply by lizzie-tersano** [reply]
Hello, sorry for the late reply. I've managed to fix the error. It was not a prisma issue, rather that nestjs was not properly converting the json body to the appropriate model. Specifically, the "quantity" attribute was not being converted from string to number. Thank you for the help, much appreciated!
